src/System/GameDataFunction.h

```diff
@@ -0,0 +1,499 @@
+#pragma once
+
+#include <basis/seadTypes.h>
+#include <math/seadVectorFwd.h>
+#include <prim/seadSafeString.h>
+#include <nn/g3d/ResFile.h>
+#include <math/seadMatrix.h>
+
+class GameDataHolderWriter;
+class ChangeStageInfo;
+class GameDataHolderAccessor;
+class ShineInfo;
+class GameDataHolder;
+class GameDataFile;
+class RiseMapPartsHolder;
+class Shine;
+class QuestInfo;
+class SessionEventProgress;
+class WorldTravelingNpc;
+class BindInfo;
+class ScenarioStartCamera;
+class CollectBgm;
+struct HackObjInfo;
+struct ShopItemInfo;
+
+namespace al {
+    class LiveActor;
+    class PlacementId;
+    class IUseSceneObjHolder;
+    class SceneObjHolder;
+    class GameDataHolderBase;
+    class LayoutActor;
+    class ActorInitInfo;
+    class PlacementInfo;
+    class Scene;
+    class IUseMessageSystem;
+    class CameraTicket;
+} // namespace al
+
+namespace GameDataFunction {
+    GameDataHolder getGameDataHolder(const al::IUseSceneObjHolder*);
+    GameDataHolder getGameDataHolder(const al::SceneObjHolder*);
+    GameDataHolder getGameDataHolder(al::GameDataHolderBase*);
+    GameDataFile getGameDataFile(GameDataHolderWriter);
```

Comment:
should probably be a pointer

Reply:
why exactly? ghidra doesn't show it as a pointer

Reply:
I mean the return type.